### PR TITLE
WIP feat(sticky): respect initial css margins

### PR DESCRIPTION
**Pull request by CanKattwinkel**

### CODE REVIEW

**Comment by CanKattwinkel** [comment]
https://w11k.github.io/angular-sticky-things/review/feature/margins/#/dev

**Comment by CanKattwinkel** [comment]
Hello @aleesaan @matthiasfritsch 

sorry for the delay of this fix, would you mind to test if your use case is finally covered? It should respect margins via input as well as css margins. #20 introduced a jump behavior for another edge case but again, served as the basis of this work - thank you so much aleesaan. I hope i got it right this time. 

And more importantly, there is now a review app for each pull request:
https://w11k.github.io/angular-sticky-things/review/feature/margins/index.html#/dev

File: projects/angular-sticky-things/src/lib/sticky-thing.directive.ts [hunk 1]
```diff
@@ -188,56 +182,81 @@ export class StickyThingDirective implements OnInit, AfterViewInit, OnDestroy {
...
+    };
   }
 
-  private makeSticky(boundaryReached: boolean = false, marginTop: number, marginBottom: number): void {
+  private prepareSticky(): void {
+
+    const marginTop = this.marginTop$.value;
+    const marginBottom = this.marginBottom$.value;
+    const stickyElStyle = this.getComputedStyle(this.stickyElement.nativeElement);
+    const boundaryElStyle = this.getComputedStyle(this.boundaryElement);
```
Comment (aleesaan) [inline projects/angular-sticky-things/src/lib/sticky-thing.directive.ts]: This will break if you have no `boundaryElement`.

**Comment by aleesaan** [comment]
Great news and thanks for getting back to us! I left a comment for something I noticed immediately while testing, as soon as you push again I'll have another look. I also noticed some missing types but since it's a WIP I assume you're still working on it - so I'll wait for your push before leaving more comments !

**Comment by aleesaan** [comment]
Also did you have a look at what I pointed out in [this issue](https://github.com/w11k/angular-sticky-things/issues/21)? I really think we should move out the `enable` logic outside the stream and just use it to un/subscribe to it, so we can avoid having the subscription (and all the related logic/triggers) running when the sticky should be disabled.

File: projects/angular-sticky-things/src/lib/sticky-thing.directive.ts (hunk 1 above)
Comment (CanKattwinkel) [reply projects/angular-sticky-things/src/lib/sticky-thing.directive.ts]: Ah, good catch, more tests and strict mode would help..

**Comment by CanKattwinkel** [comment]
Yes, Issue 21 is on my radar! I'll do this feature, then some stuff for developer experience and then number 21.

**Comment by CanKattwinkel** [comment]
I'll merge into dev since I developed some CI features along with this feature. I'll wait with the release to master though, maybe some issue comes up.

**Comment by aleesaan** [comment]
Hi Can,

I just tested the package with my app again but the issue I was fixing with PR #20 is still there...
